grails-web-gsp/src/main/groovy/org/grails/web/gsp/GroovyPagesTemplateRenderer.java

```diff
@@ -115,6 +115,10 @@ public void render(GrailsWebRequest webRequest, TemplateVariableBinding pageScop
         final Object controller = webRequest.getAttribute(GrailsApplicationAttributes.CONTROLLER, GrailsWebRequest.SCOPE_REQUEST);
         Template t = findAndCacheTemplate(controller, pageScope, templateName, contextPath, pluginName, uri);
         if (t == null) {
+            String optional = getStringValue(attrs, "optional");
+            if (Boolean.parseBoolean(optional.isBlank()? "false" : optional)) {
+                return; // allow missing templates if optional == "true"
+            }
```

Comment:
How about extracting this to a method `getBooleanValue` next to `getStringValue`:

```java
private String getStringValue(Map<String, Object> attrs, String key) {
    Object val = attrs.get(key);
    if (val == null) return "";
    return String.valueOf(val);
}

private boolean getBooleanValue(Map<String, Object> attrs, String key) {
    String val = getStringValue(attrs, key);
    if (val.isBlank()) return false;
    return Boolean.parseBoolean(val);
}
```

Then we can get:

```java
if (getBooleanValue(attrs, "optional")) {
    return; // allow missing templates if optional == "true"
}
```

Reply:
done